sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,280 @@
+        },
+        order: {
+          msfaaNumber: "ASC",
+        },
+      });
+      expect(msfaaUpdatedRecords).toHaveLength(msfaaInputData.length);
+      const [reactivatedMSFAA] = msfaaUpdatedRecords;
+      // Validate reactivated confirmed record.
+      expect(reactivatedMSFAA.dateSigned).toBe("2023-05-02");
+      expect(reactivatedMSFAA.serviceProviderReceivedDate).toBe("2023-05-03");
+      expect(reactivatedMSFAA.cancelledDate).toBe(null);
+    });
+
+    it("Should reactivate a cancelled MSFAA when the same MSFAA is received in the response file and re-associate this reactivated MSFAA with all pending disbursements.", async () => {
+      // Arrange
+      // Create a cancelled MSFAA. The Msfaa number used for creating the cancelled Msfaa record is the same as the one used in the msfaa-full-time-file-with-reactivation-record.dat
+      // Ensuring that any previous runs of this test or any other test do not have the same Msfaa id as the one in the re-activation file.
+      await db.msfaaNumber.update(
+        { msfaaNumber: MSFAA_FULL_TIME_MARRIED.msfaaNumber },
+        { msfaaNumber: THROW_AWAY_MSFAA_NUMBER },
+      );
+      const msfaaInputData = [MSFAA_FULL_TIME_MARRIED];
+      const student = await saveFakeStudent(db.dataSource);
+      const createdMSFAARecords = await saveMSFAATestInputsData(
+        db,
+        msfaaInputData,
+        { student },
+        {
+          state: MSFAAStates.CancelledOtherProvince,
+          offeringIntensity: OfferingIntensity.fullTime,
+        },
+      );
+      //Create Pending MSFAA and associate it with disbursements from the two applications.
+      const currentMSFAA = createFakeMSFAANumber(
+        { student },
+        {
+          state: MSFAAStates.Pending,
+          offeringIntensity: OfferingIntensity.fullTime,
+        },
+      );
+      await db.msfaaNumber.save(currentMSFAA);
+      // Create 2 applications with 4 disbursements - 3 pending state and 1 sent.
+      const applicationA = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        {
+          student,
+          msfaaNumber: currentMSFAA,
+        },
+        {
+          applicationStatus: ApplicationStatus.Completed,
+          offeringIntensity: OfferingIntensity.fullTime,
+          createSecondDisbursement: true,
+          disbursementScheduleStatusA: DisbursementScheduleStatus.Sent,
+          disbursementScheduleStatusB: DisbursementScheduleStatus.Pending,
+        },
+      );
+      const applicationB = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        {
+          student,
+          msfaaNumber: currentMSFAA,
+        },
+        {
+          applicationStatus: ApplicationStatus.Completed,
+          offeringIntensity: OfferingIntensity.fullTime,
+          createSecondDisbursement: true,
+          disbursementScheduleStatusA: DisbursementScheduleStatus.Pending,
+          disbursementScheduleStatusB: DisbursementScheduleStatus.Sent,
```

Comment:
as per the current logic, 2nd disbursement will only be sent after the first one. correct me if I am wrong @andrewsignori-aot @guru-aot @dheepak-aot @andrepestana-aot

Reply:
You are right @andrewsignori-aot.

Reply:
I noticed that `disbursementScheduleStatusB` is deleted, but again it's the old logic right? which is incorrect right?